app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx

```diff
@@ -193,117 +202,140 @@ export function ProtocolRunHeader({
   }
 
   return (
-    <Flex
-      ref={protocolRunHeaderRef}
-      backgroundColor={COLORS.white}
-      border={BORDERS.lineBorder}
-      borderRadius={BORDERS.radiusSoftCorners}
-      flexDirection={DIRECTION_COLUMN}
-      gridGap={SPACING.spacing16}
-      marginBottom={SPACING.spacing16}
-      padding={SPACING.spacing16}
-    >
-      {showAnalysisErrorModal &&
-        analysisErrors != null &&
-        analysisErrors?.length > 0 && (
-          <ProtocolAnalysisErrorModal
-            displayName={displayName}
-            errors={analysisErrors}
-            onClose={handleErrorModalCloseClick}
-            robotName={robotName}
-          />
-        )}
+    <>
+      {showRunFailedModal ? (
+        <RunFailedModal
+          robotName={robotName}
+          runId={runId}
+          setShowRunFailedModal={setShowRunFailedModal}
+          highestPriorityError={highestPriorityError}
+        />
+      ) : null}
+      <Flex
+        ref={protocolRunHeaderRef}
+        backgroundColor={COLORS.white}
+        border={BORDERS.lineBorder}
+        borderRadius={BORDERS.radiusSoftCorners}
+        flexDirection={DIRECTION_COLUMN}
+        gridGap={SPACING.spacing16}
+        marginBottom={SPACING.spacing16}
+        padding={SPACING.spacing16}
+      >
+        {showAnalysisErrorModal &&
+          analysisErrors != null &&
+          analysisErrors?.length > 0 && (
```

Comment:
do you need this extra null protection?

```suggestion
          analysisErrors.length > 0 && (
```